grails-app/services/com/unifina/service/CanvasService.groovy

```diff
@@ -201,8 +202,50 @@ class CanvasService {
 		return grailsLinkGenerator.link(controller: 'canvas', action: 'editor', id: canvas.id, absolute: true)
 	}
 
+	@CompileStatic
+	def addExampleCanvases(SecUser user, List<Canvas> examples) {
+		for (Canvas c : examples) {
+			switch (c.exampleType) {
+				// Create a copy of the example canvas for the user and grant all permissions.
+				case ExampleType.COPY:
+					Canvas canvas = new Canvas()
+					canvas.dateCreated = c.dateCreated
+					canvas.lastUpdated = c.lastUpdated
+					canvas.name = c.name
+					canvas.json = c.json
```

Comment:
We should write `canvas.json` **after** the `resetUiChannels`, reconstruction, and `ensureUiChannel` calls